compiler/rustc_trait_selection/src/traits/mod.rs

```diff
@@ -143,9 +143,10 @@ pub fn type_known_to_meet_bound_modulo_regions<'tcx>(
     ty: Ty<'tcx>,
     def_id: DefId,
     span: Span,
+    rest: &[ty::GenericArg<'tcx>],
```

Comment:
`rest` feels dangerous, can you instead use `mk_substs_trait_with_effects` in this function?